It might be a local problem, but I repeatedly get:
```
/nix/store/wn6slnv6m5glx8lm6gkpjr1h3bxrrs67-dmd-2.067.1.drv

[...]
 --- std.socket(521) test fails depending on environment ---
 (core.exception.AssertError@std/socket.d(523): Assertion failure
----------------
generated/linux/release/64/unittest/libphobos2-ut.so(+0x3bce724) [0x7ffff695a724]
generated/linux/release/64/unittest/libphobos2-ut.so(nothrow @safe void std.socket.__unittestL518_3().__lambda1()+0x3d) [0x7ffff6929375]
generated/linux/release/64/unittest/libphobos2-ut.so(@trusted void std.socket.softUnittest(@safe void delegate(), int)+0x23) [0x7ffff6928d5b]
generated/linux/release/64/unittest/libphobos2-ut.so(@safe void std.socket.__unittestL518_3()+0x18) [0x7ffff6929330]
generated/linux/release/64/unittest/libphobos2-ut.so(void std.socket.__modtest()+0x9) [0x7ffff695a621]
generated/linux/release/64/unittest/test_runner() [0x40141c]
generated/linux/release/64/unittest/libphobos2-ut.so(runModuleUnitTests+0xe1) [0x7ffff74322a9]
generated/linux/release/64/unittest/libphobos2-ut.so(void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll()+0x17) [0x7ffff7454b17]
generated/linux/release/64/unittest/libphobos2-ut.so(void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate())+0x2d) [0x7ffff7454acd]
generated/linux/release/64/unittest/libphobos2-ut.so(_d_run_main+0x1ec) [0x7ffff7454a44]
generated/linux/release/64/unittest/test_runner() [0x40160c]
/nix/store/q0prbrw9n7h2sjqmla8ac32lfyq0w91z-glibc-2.25-49/lib/libc.so.6(__libc_start_main+0xf0) [0x7ffff22d4560])
No service for epmap.
****** FAIL release64 std.socket
core.exception.RangeError@std/socket.d(865): Range violation
----------------
generated/linux/release/64/unittest/libphobos2-ut.so(+0x3bce6f4) [0x7ffff695a6f4]
generated/linux/release/64/unittest/libphobos2-ut.so(@safe void std.socket.__unittestL860_5()+0x59) [0x7ffff6929bc9]
generated/linux/release/64/unittest/libphobos2-ut.so(void std.socket.__modtest()+0x13) [0x7ffff695a62b]
generated/linux/release/64/unittest/test_runner() [0x40141c]
generated/linux/release/64/unittest/libphobos2-ut.so(runModuleUnitTests+0xe1) [0x7ffff74322a9]
generated/linux/release/64/unittest/libphobos2-ut.so(void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll()+0x17) [0x7ffff7454b17]
generated/linux/release/64/unittest/libphobos2-ut.so(void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate())+0x2d) [0x7ffff7454acd]
generated/linux/release/64/unittest/libphobos2-ut.so(_d_run_main+0x1ec) [0x7ffff7454a44]
generated/linux/release/64/unittest/test_runner() [0x40160c]
/nix/store/q0prbrw9n7h2sjqmla8ac32lfyq0w91z-glibc-2.25-49/lib/libc.so.6(__libc_start_main+0xf0) [0x7ffff22d4560]
make: *** [posix.mak:376: unittest/std/socket.d] Error 1
```

but it seems the same before this PR as well...

--- a/pkgs/development/compilers/dmd/2.067.1.nix
+++ b/pkgs/development/compilers/dmd/2.067.1.nix
@@ -101,7 +101,8 @@ stdenv.mkDerivation rec {
       cd ..
   '';
 
-  doCheck = true;
+  # disable check phase because some tests are not working with sandboxing
+  doCheck = false;
 
   checkPhase = ''
       cd dmd

--- a/pkgs/development/compilers/dmd/default.nix
+++ b/pkgs/development/compilers/dmd/default.nix
@@ -97,7 +97,8 @@ stdenv.mkDerivation rec {
       cd ..
   '';
 
-  doCheck = true;
+  # disable check phase because some tests are not working with sandboxing
+  doCheck = false;
 
   checkPhase = ''
       cd dmd

--- a/pkgs/development/compilers/ldc/default.nix
+++ b/pkgs/development/compilers/ldc/default.nix
@@ -81,7 +81,8 @@ stdenv.mkDerivation rec {
 
   makeFlags = [ "DMD=$DMD" ];
 
-  doCheck = true;
+  # disable check phase because some tests are not working with sandboxing
+  doCheck = false;
 
   checkPhase = ''
       ctest -j $NIX_BUILD_CORES -V DMD=$DMD

